validator/docs/building_a_command_line_amp_validator_for_mac_os_x.md

```diff
@@ -0,0 +1,181 @@
+#Building a command-line AMP Validator: Mac OS X
+
+## Purpose
+
+This documents how to build the official AMP-HTML command-line AMP Validator on Mac OS X. Once built in this fashion it can then be executed from a OS X terminal session to validate candidate HTML files over the local file system or the web as follows:
+
+No HTML file passed to the program:
+
+```
+$ node validate
+usage: validate <file.html or url>
+```
+
+Valid HTML file passed to the program:
+
+```
+$ node validate testdata/feature_tests/minimum_valid_amp.html
+PASS
+```
+
+Invalid HTML file passed to the program:
+
+```
+$ node validate testdata/feature_tests/empty.html
+FAIL
+empty.html:1:0 The mandatory tag 'html ⚡ for top-level html' is missing or incorrect. (see [https://github.com/ampproject/amphtml/blob/master/spec/amp-html-format.md#ampd])
+```
+
+## References
+- Development on AMP HTML 
+    - [https://github.com/ampproject/amphtml/blob/master/DEVELOPING.md](https://github.com/ampproject/amphtml/blob/master/DEVELOPING.md) 
+- AMP HTML ⚡ Validator 
+    - [https://github.com/ampproject/amphtml/blob/master/validator/README.md](https://github.com/ampproject/amphtml/blob/master/validator/README.md) 
+- Validating outside the browser - command line tool #937 
+    - [https://github.com/ampproject/amphtml/issues/937](https://github.com/ampproject/amphtml/issues/937) 
+
+## Dependencies and configuration
+
+### Homebrew
+
+#### Installing Homebrew
+
+See: [http://brew.sh/](http://brew.sh/)
+
+```
+$ ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+```
+
+#### Maintaining your Homebrew environment
+
+See GitHub Gist: [pietergreyling/mybrew.sh](https://gist.github.com/pietergreyling/43b00966f0a775a84ac8)
+  
+```
+#!/bin/bash
+#- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+#-- desc: To clean up and verify your Homebrew setup. 
+#-- exec: In a terminal session run: $ ./mybrew.sh
+clear
+echo "-- Updating your Homebrew installation at: " && brew --prefix
+sudo chown -R $(whoami):admin /usr/local
+brew update && brew upgrade
+echo "-- \$HOMEBREW_INSTALL: $HOMEBREW_INSTALL"
+brew prune
+brew doctor
+#- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+```
+
+### Node JS
+
+```
+$ brew install node
+```
+
+### Edit /etc/hosts
```

Comment:
I'd like to drop this section since it's not needed for the validator. We have a [DEVELOPMENT.md](https://github.com/ampproject/amphtml/blob/master/DEVELOPING.md#installation) with these instructions instead.